src/pages/en/guides/migration-guide/from-gatsby.md

```diff
@@ -0,0 +1,1184 @@
+
+```jsx title="layout.jsx"
+interface LayoutProps {
+    title: string;
+}
+
+const Layout = ({title}: LayoutProps) => {
+    // ...
+}
+```
+
+In Astro, reuse the TypeScript interface and type cast the `Astro.props` destructure to enforce type safety:
+
+```astro title="Layout.astro"
+---
+interface LayoutProps {
+    title: string;
+}
+
+const {title} = Astro.props as LayoutProps;
+---
+
+<!-- ... -->
+```
+
+### Gatsby Styling to Astro
+
+Styling in both Gatsby and Astro comes in a few different flavors:
+
+- Inline styles
+- Component-specific styling
+- Global styling
+
+#### Inline Styles
+
+Convert any inline style objects in React (`style={{fontWeight: "bold", }}`) to inline HTML style attributes (`style="font-weight:bold;"`).
+
+```jsx
+// JSX
+<p style={{fontWeight: "bold"}}>Hello, world</p>
+```
+
+```astro
+// Astro
+<p style="font-weight: bold;">Hello, world</p>
+```
+
+#### Component-specific styling
+
+One of Astro's unique capabilities enables you to use standard `<style>` tags for scoped CSS styles.
+
+```astro
+<p>Hello, world</p>
+
+<!-- The following will only apply to this component -->
+<style>
+p {
+   font-weight: bold;
+}
+</style>
+```
+
+However, you can extract your styling to a [CSS Module file](https://docs.astro.build/en/guides/imports/#css-modules) and import that instead.
+
+```astro title="src/components/Hello.astro"
+---
+import styles from './Hello.module.css';
+---
+
+<p>Hello, world</p>
+```
+
+ ```css title="src/components/Hello.module.css"
+ p {
+    font-weight: bold;
+ }
+ ```
+
+#### Global Styling
+
+Global styling is applied similarly between Gatsby and Astro by importing a `css` file within a layout file.
+
+```astro title="src/layouts/Layout.astro" {2}
+---
+import "global.css";
+---
+
+<html>
+  <head>
+    <meta charset="utf-8" />
+	<link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+	<meta name="viewport" content="width=device-width" />
+	<meta name="generator" content={Astro.generator} />
+	<title>Astro</title>
+  </head>
+  <body>
+    <slot/>
+  </body>
+</html>
+```
+
+```css title="src/layouts/global.css"
+p {
+    font-weight: bold;
+}
+```
+
+#### CSS Preprocessors
+
+Astro supports installing and using many [CSS preprocessors](/en/guides/styling/#css-preprocessors).
+
+As an example, to use Sass, you would run:
+
+```shell
+npm install -D sass
+```
+
+After doing so, you're then able to import `.scss` or `.sass` files as you would a `.css` file. Additionally, you're able to declare the CSS preprocessor used to modify `style` tags:
+
+```astro title="src/layouts/Layout.astro"
+<p>Hello, world</p>
+
+<style lang="scss">
+p {
+   color: black;
+   
+   &:hover {
+       color: red;
+   }
+}
+</style>
+```
+
+### Gatsby Code Comments to Astro
+
+An Astro file uses JavaScript code comments in the frontmatter `//` but HTML code comments in the template body. (`<!-- -->`)
```

Comment:
Should we include something like the code comment sample that's here? https://docs.astro.build/en/core-concepts/astro-components/#component-structure

![Screenshot 2022-12-12 13 36 06](https://user-images.githubusercontent.com/5098874/207114678-176b886a-cb45-4f53-8f42-b95e62f59b80.png)